st2common/st2common/bootstrap/triggersregistrar.py

```diff
@@ -0,0 +1,182 @@
+
+        return registered_count
+
+    def register_triggers_from_pack(self, pack_dir):
+        """
+        Register all the triggers from the provided pack.
+
+        :return: Number of triggers registered.
+        :rtype: ``int``
+        """
+        pack_dir = pack_dir[:-1] if pack_dir.endswith('/') else pack_dir
+        _, pack = os.path.split(pack_dir)
+        triggers_dir = self._pack_loader.get_content_from_pack(pack_dir=pack_dir,
+                                                               content_type='triggers')
+
+        # Register pack first
+        self.register_pack(pack_name=pack, pack_dir=pack_dir)
+
+        registered_count = 0
+        if not triggers_dir:
+            return registered_count
+
+        LOG.debug('Registering triggers from pack %s:, dir: %s', pack, triggers_dir)
+
+        try:
+            triggers = self._get_triggers_from_pack(triggers_dir=triggers_dir)
+            registered_count = self._register_triggers_from_pack(pack=pack, triggers=triggers)
+        except Exception as e:
+            if self._fail_on_failure:
+                raise e
+
+            LOG.exception('Failed registering all triggers from pack "%s": %s', triggers_dir,
+                          str(e))
+
+        return registered_count
+
+    def _get_triggers_from_pack(self, triggers_dir):
+        return self.get_resources_from_pack(resources_dir=triggers_dir)
+
+    def _register_triggers_from_pack(self, pack, triggers):
+        registered_count = 0
+
+        for trigger in triggers:
+            try:
+                self._register_trigger_from_pack(pack=pack, trigger=trigger)
+            except Exception as e:
+                if self._fail_on_failure:
+                    raise e
+
+                LOG.debug('Failed to register trigger "%s": %s', trigger, str(e))
+            else:
+                LOG.debug('Trigger "%s" successfully registered', trigger)
+                registered_count += 1
+
+        return registered_count
+
+    def _register_trigger_from_pack(self, pack, trigger):
+        trigger_metadata_file_path = trigger
+
+        LOG.debug('Loading trigger from %s.', trigger_metadata_file_path)
+        content = self._meta_loader.load(file_path=trigger_metadata_file_path)
+
+        pack_field = content.get('pack', None)
+        if not pack_field:
+            content['pack'] = pack
+            pack_field = pack
+        if pack_field != pack:
+            raise Exception('Model is in pack "%s" but field "pack" is different: %s' %
+                            (pack, pack_field))
+
+        entry_point = content.get('entry_point', None)
+        if not entry_point:
+            raise ValueError('Trigger definition missing entry_point')
+
+        triggers_dir = os.path.dirname(trigger_metadata_file_path)
+        trigger_file_path = os.path.join(triggers_dir, entry_point)
+        artifact_uri = 'file://%s' % (trigger_file_path)
+        content['artifact_uri'] = artifact_uri
```

Comment:
TriggerType DB model doesn't take those two attributes so you should remove them.